packages/components/pie-modal/test/component/pie-modal.spec.ts

```diff
@@ -26,779 +31,763 @@ test.describe('modal', () => {
+                    const component = renderTestPieModal({
+                        returnFocusAfterCloseSelector: '[data-test-id="focus-me"]',
+                    });
+
+                    await mount(WebComponentTestWrapper, {
+                        props: {
+                            pageMode: true,
+                        },
+                        slots: {
+                            component,
+                            pageMarkup: `<div>
+                                <button id="default"></button>
+                                <button data-test-id="focus-me" id="actual-focus"></button>
+                                <button data-test-id="focus-me"></button>
+                            </div>`,
+                        },
+                    });
+
+                    // Act
+                    await modalPage.clickCloseModal();
+
+                    const focusedElement = await page.locator(':focus');
+                    const focusedElementId = await focusedElement.getAttribute('id');
+
+                    // Assert
+                    expect(focusedElementId).toBe('actual-focus');
+                });
+            });
+
+            test.describe('when not given', () => {
+                [{
+                    mechanism: 'close button',
+                    modalCloseFunction: async (page : Page) => {
+                        const modalPage = new PieModalPage(page);
+                        await modalPage.clickCloseModal();
+                    },
+                }, {
+                    mechanism: 'Esc key',
+                    modalCloseFunction: async (page : Page) => {
+                        await page.keyboard.press('Escape');
+                    },
+                }].forEach(({ mechanism, modalCloseFunction }) => {
+                    test.describe(`and closed by the ${mechanism}`, () => {
+                        test('should return focus to the element that opens the modal', async ({ page, mount }) => {
+                            // Arrange
+                            const component = renderTestPieModal({ isOpen: false });
+
+                            await mount(WebComponentTestWrapper, {
+                                props: {
+                                    pageMode: true,
+                                },
+                                slots: {
+                                    component,
+                                    pageMarkup: `<div>
+                                        <button id="not-me"></button>
+                                        <button data-test-id="open-modal" id="default"></button>
+                                    </div>`,
+                                },
                             });
-                        });
 
-                        // Act
-                        await page.click('[data-test-id="open-modal"]');
-                        await modalCloseFunction(page);
+                            await page.evaluate(() => {
```

Comment:
This can probably be refactored to use the `on` event code like we do in other tests.

Reply:
I don't think we want to do that as it would mean adding defs for a single test that is specific to just this implementation?

Reply:
As discussed, this code is fine :)